packages/components/src/components/hds/table/th-selectable.ts

```diff
@@ -115,10 +100,4 @@ export default class HdsTableThSelectable extends Component<HdsTableThSelectable
       onSelectionChange(target, this.args.selectionKey);
     }
   }
-
-  updateAriaLabel(event: Event): void {
-    // Assert event.target as HTMLInputElement to access the 'checked' property
-    const target = event.target as HdsFormCheckboxBaseSignature['Element'];
-    this.isSelected = target.checked;
```

Comment:
Its safe to remove this line because the state is already updated in the `onSelectionChange` function.

Reply:
Isn't this also the case for the AdvancedTable?

Reply:
Yes, I updated it in this pr: https://github.com/hashicorp/design-system/pull/2688

Reply:
Nice! Thank you!